packages/kolme/src/core/kolme.rs

```diff
@@ -1190,15 +1214,23 @@ impl<App: KolmeApp> Kolme<App> {
     }
 
     async fn init_remote_data_listener(&self) -> Result<()> {
-        self.inner.store.init_remote_data_listener().await?;
-        let mut sub = self.inner.store.subscribe_remote_data();
-        let kolme = self.clone();
+        let Some(mut listener) = self.inner.store.listen_remote_data().await? else {
+            return Ok(());
+        };
+        let weak_kolme = self.weak();
         tokio::spawn(async move {
             loop {
-                sub.listen().await;
-                tracing::info!("Resyncing after new remote data notification");
-                if let Err(err) = kolme.resync().await {
-                    tracing::error!("Error resyncing after remote data notification: {err:?}");
+                let resync = tokio::time::timeout(Duration::from_secs(1), listener.recv())
```

Comment:
What's the purpose of the timeout? Is it to ensure we shut down when the Kolme instance we're using is no longer active?

Reply:
I'd like to avoid a loop like this if possible. An alternative approach:

* Include some new StopListeningOnDrop value inside the Kolme value
* On drop, it stops the listener entirely
* Receiving on a stopped listener returns a ListenerStopped error value (separate from other errors that may occur), which causes this loop to exit

Is something like that possible?

Reply:
I've pushed an implementation of this approach.  I had to add Drop implementation for Store that explicitly closes the pool, otherwise the PgListener holds the pool open.

A different approach to avoiding the timeout would be to put a JoinSet containing the long-lived task in KolmeInner.  When the JoinSet is dropped, it will abort the task (which will drop the PgListener).